GVFS/GVFS.Common/LibGit2RepoPool.cs

```diff
@@ -90,6 +140,20 @@ private void ReturnToPool(LibGit2Repo repo)
             }
         }
 
+        private void ResetRepoDisposalTimer()
+        {
+            this.repoDisposalTimer.Change(this.repoDisposalDueTime, this.repoDisposalPeriod);
+        }
+
+        private void TryDropARepo()
+        {
+            if (this.pool.TryTake(out LibGit2Repo repo, TryTakeNoWait))
+            {
+                repo.Dispose();

```

Comment:
What happens if the packfile maintenance step fails because there are still some repos in the pool that have not been disposed?  Will VFS4G wait for the entire packfile maintenance period before it tries again?

Reply:
The step doesn't actually _fail_, but it leaves some .idx files around without corresponding .pack files. The next time it is run, it will clean up those files (assuming we had some period where the libgit2 repo was disposed and released the handle).

When we delete the .pack associated with the .idx, new libgit2 repos will not hold a handle to the .idx files because they will notice there isn't a paired .pack.